psyneulink/core/globals/utilities.py

```diff
@@ -440,7 +440,8 @@ def iscompatible(candidate, reference=None, **kargs):
     try:
         with warnings.catch_warnings():
             warnings.simplefilter(action='ignore', category=FutureWarning)
-            if reference is not None and (candidate == reference):
+            # np.array(...).size > 0 checks for empty list. Everything else create single element (dtype=obejct) array
+            if reference is not None and np.array(candidate, dtype=object).size > 0 and (candidate == reference):
```

Comment:
No issue with this, but we could potentially also use
```suggestion
            if reference is not None and np.array(candidate, dtype=object).size > 0 and safe_equals(candidate, reference):
```

although safe_equals would need a couple small changes because I think the type of warning produced by that == check has changed since it was written

Reply:
I had the same idea [0,1], but it gave enough errors to leave it for the next time.
Most of the remaining warnings (in total number) are from the elementwise comparison and np.matrix so it's definitely worth revisting.

[0] https://github.com/jvesely/PsyNeuLink/actions/runs/3492618053
[1] https://ci.appveyor.com/project/jvesely/psyneulink-cuda/builds/45417908/job/m8adfgaec8pfs27u

Reply:
Ah ok, makes sense. I can make those changes separately then, they're not many.